libraries/_helper.rb

```diff
@@ -433,12 +456,56 @@ def ensure_update_center_present!
         headers = {
           'Accept' => 'application/json'
         }
-        http = Net::HTTP.new(uri.host, uri.port)
-        http.use_ssl = true if uri.scheme == 'https'
-        response = http.post(uri.path, extracted_json, headers)
+        http = Chef::HTTP.new(uri)
+        http.post(uri, extracted_json, headers)
 
         true
       end
     end
+
+    #
+    # Idempotently create password for the keystore
+    #
+    def keystore_pass!
+      node.run_state[:jenkins_keystore_pass] ||= SecureRandom.uuid
```

Comment:
@schisamo That's correct. `create_keystore!` is going to remove the old keystore and make a new one every run too. Seems like the password for that keystore should also be only stored in the run state.